src/http-api/fetch.ts

```diff
@@ -301,17 +301,22 @@ export class FetchHttpApi<O extends IHttpOpts> {
         return res as ResponseType<T, O>;
     }
 
-    private clearUrlParamsForLogs(url: URL | string): string {
+    private sanitizeUrlForLogs(url: URL | string): string {
         try {
             let asUrl: URL;
             if (typeof url === "string") {
                 asUrl = new URL(url);
             } else {
                 asUrl = url;
             }
-            // get just the path to remove any potential url param that could have
-            // some potential secrets
-            return asUrl.origin + asUrl.pathname;
+            // Remove the values of any URL params that could contain potential secrets
+            const sanitizedQs = new URLSearchParams();
+            for (const key of asUrl.searchParams.keys()) {
+                sanitizedQs.append(key, "xxx");
+            }
+            const sanitizedQsUrlPiece = sanitizedQs.toString() ? `?${sanitizedQs.toString()}` : "";
```

Comment:
I think calling `toString` twice here probably actually does the work of converting to a string twice, which seems a bit of a waste.

Reply:
Updated to only stringify once :fast_forward: